### DISCUSSION TITLE
Guidance for setting a custom WITH clause on my table?

### DISCUSSION THREAD

**Discussion by blockjon-dd** [sqlalchemy/sqlalchemy#9323] [Usage Questions] [answered]
I'm using a Postgres-style database called CockroachDB which has an option I'd like to use (documented [here](https://www.cockroachlabs.com/docs/stable/row-level-ttl.html#examples))

I'm trying to figure out how to customize a WITH clause as seen in the example below:
```
CREATE TABLE ttl_test_ttl_expiration_expression (
  id INT PRIMARY KEY,
  expire_at TIMESTAMPTZ
) WITH (ttl_expiration_expression = 'expire_at');
```
When I define my SqlAlchemy class, what can I do in order to populate the with clause as seen in the example above?

**Comment by zzzeek** [comment]
cc @gordthompson

**Comment by blockjon-dd** [comment]
I'm guessing it might be this feature here:
https://docs.sqlalchemy.org/en/14/dialects/postgresql.html#index-storage-parameters

**Reply by CaselIT** [reply]
no, that's for the index creation, so when running "CREATE INDEX ..."

**Comment by CaselIT** [comment, marked as answer]
The supported table options by the postgresql dialect are https://docs.sqlalchemy.org/en/14/dialects/postgresql.html#postgresql-table-options that does not seem to support open ended options.

You should probably use the 3rd party dialect for CockroachDB https://pypi.org/project/sqlalchemy-cockroachdb/ that may build on these to add what you are looking for

**Reply by blockjon-dd** [reply]
Thank you @CaselIT.
